sound/soc/sof/sof-ipc-test-client.c

```diff
@@ -6,22 +6,271 @@
 //
 
 #include <linux/completion.h>
+#include <linux/debugfs.h>
+#include <linux/ktime.h>
 #include <linux/mod_devicetable.h>
 #include <linux/module.h>
 #include <linux/pm_runtime.h>
+#include <linux/slab.h>
+#include <linux/uaccess.h>
 #include <linux/virtual_bus.h>
+#include <sound/sof/header.h>
 #include "sof-client.h"
 
+#define MAX_IPC_FLOOD_DURATION_MS 1000
+#define MAX_IPC_FLOOD_COUNT 10000
+#define IPC_FLOOD_TEST_RESULT_LEN 512
 #define SOF_IPC_CLIENT_SUSPEND_DELAY_MS 3000
 
-/* TODO: dummy probe/remove callbacks for now */
+struct sof_ipc_client_data {
+	struct dentry *dfs_root;
+	char *buf;
+};
+
+static int sof_debug_ipc_flood_test(struct sof_client_dev *cdev,
+				    bool flood_duration_test,
+				    unsigned long ipc_duration_ms,
+				    unsigned long ipc_count)
+{
+	struct sof_ipc_client_data *ipc_client_data = cdev->data;
+	struct device *dev = &cdev->vdev.dev;
+	struct sof_ipc_cmd_hdr hdr;
+	struct sof_ipc_reply reply;
+	u64 min_response_time = U64_MAX;
+	ktime_t start, end, test_end;
+	u64 avg_response_time = 0;
+	u64 max_response_time = 0;
+	u64 ipc_response_time;
+	int i = 0;
+	int ret;
+
+	/* configure test IPC */
+	hdr.cmd = SOF_IPC_GLB_TEST_MSG | SOF_IPC_TEST_IPC_FLOOD;
+	hdr.size = sizeof(hdr);
+
+	/* set test end time for duration flood test */
+	if (flood_duration_test)
+		test_end = ktime_get_ns() + ipc_duration_ms * NSEC_PER_MSEC;
+
+	/* send test IPC's */
+	while (1) {
```

Comment:
This doesnt even fit in a single line with 80 chars. The reason I used a while (1) is because if either the requested count or the time were to be 0, you'd never enter this function. So you'd atleast run this loop once and the check at the end of the block is quite intuitive.

Reply:
and the i < ipc_count would be incorrect to test for the duration test

Reply:
@ranj063 that's why I didn't propose to use `i < ipc_count` but `i < ipc_count || (flood_duration_test && ktime_to_ns(end) < test_end)` instead. As for 80 chars - firstly I would split that e.g. as
```
for (i = 0;
    i < ipc_count || (flood_duration_test && ktime_to_ns(end) < test_end);
    i++) {
```
But in general it certainly isn't a show-stopper, just a personal preference. I find loops, that have a clear initialisation, condition, and step to be clearer as `for` but it's certainly a matter of personal judgement. Extreme cases like literal
```
i = 0;
while (1) {
if (i >= 10) break;
...
i++;
}
```
must certainly be prosecuted, but milder cases are moving to the area of personal taste IMHO :-)

Reply:
ok I've change it to a for now with a slight change in the condition.